flake.nix

```diff
@@ -0,0 +1,60 @@
+{
+  description = "A Nix-flake-based development environment for command-server";
+
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
```

Comment:
is there a reason we use `unstable` here?

Reply:
The only stable channels correspond to NixOS releases. It seems unlikely we want to pin everything to a NixOS release? I don't think the nixos channels are tested on macOS. https://nixos.wiki/wiki/Nix_channels

Reply:
Fine by me; as mentioned, I know basically nothing about nix, so just saw the word "unstable" and thought it might be worth flagging 😅